### PR TITLE
Fix Raw String of non-literal

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR fixes a bug where the 'raw string' expression would return the formatted string of a non-literal

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/5515

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprRawString.java [hunk 1]
```diff
@@ -46,6 +45,8 @@
 @Since("2.7")
 public class ExprRawString extends SimpleExpression<String> {
 
+	private static final Pattern HEX_PATTERN = Pattern.compile("&x((?:&\\p{XDigit}){6})");
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprRawString.java:48]: You can use Utils's hex pattern field, just make it public
https://github.com/SkriptLang/Skript/blob/aba2f87adde20d2ca2fbafc2d0eefc90faafd12a/src/main/java/ch/njol/skript/util/Utils.java#L643

File: src/main/java/ch/njol/skript/expressions/ExprRawString.java [hunk 2]
```diff
@@ -71,14 +72,21 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
...
 		for (Expression<? extends String> message : messages) {
 			if (message instanceof VariableString) {
-				strings.add(((VariableString) message).toUnformattedString(e));
+				strings.add(((VariableString) message).toUnformattedString(event));
 				continue;
 			}
-			strings.addAll(Arrays.asList(message.getArray(e)));
+			for (String string : message.getArray(event)) {
+				String raw = SkriptColor.replaceColorChar(string);
+				if (raw.contains("&x")) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/expressions/ExprRawString.java:84]: Even tho `&x` is hardcoded within Skript but for comparison it should be made on lowercased `raw`

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprRawString.java (hunk 1 above)
Comment (UnderscoreTud) [reply src/main/java/ch/njol/skript/expressions/ExprRawString.java:48]: That's not the pattern I'm looking for. The pattern you mentioned matches `#abcdef` and the pattern I'm trying to match is `&x&a&b&c&d&e&f` (The spigot way of representing a hex code)

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/expressions/ExprRawString.java:48]: Okay. Maybe add it to Utils as well like `HEX_PATTERN_SPIGOT`

**Review by TheLimeGlass — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]

**Pull request merged**